lib/models/update-checker.js

```diff
@@ -64,23 +64,30 @@ UpdateChecker.prototype.doCheck = function() {
 };
 
 UpdateChecker.prototype.checkNPM = function() {
-  // make an http call to npm to get the latest version
-  var http = require('http');
-  var concat = require('concat-stream');
+  // use npm to determine the currently availabe ember-cli version
+  var npm = require('npm'),
+      npmViewConfig = {
+        'loglevel': 'silent',
+        'fetch-retries': 1,
+        'cache-min': 1,
+        'cache-max': 0
+      };
 
   return new Promise(function(resolve, reject) {
-    http.get('http://registry.npmjs.org/ember-cli/latest', function(res) {
-      res.setEncoding('utf8');
-
-      res.pipe(concat(function (data) {
-        try {
-          resolve(JSON.parse(data).version);
-        } catch(error) {
-          reject(error);
-        }
-      }));
-
-      res.on('error', reject);
+    npm.load(npmViewConfig, function(err) {
```

Comment:
we can use `denodeify` https://github.com/ember-cli/ember-cli/blob/master/lib/ext/promise.js#L13-L21
docs: https://github.com/tildeio/rsvp.js/blob/master/lib/rsvp/node.js#L73-L201